test/Microsoft.AspNetCore.Server.KestrelTests/PipelineExtensionTests.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO.Pipelines;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.Kestrel.Internal.Http;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.KestrelTests
+{
+    public class PipelineExtensionTests : IDisposable
```

Comment:
I added a test for everything in the ASCII range, 0x00 - 0x7F. As mentioned here https://github.com/aspnet/KestrelHttpServer/issues/1508#issuecomment-289083405, characters outside the ASCII range should never reach this point...in theory.